brion/nodes.cpp

```diff
@@ -0,0 +1,114 @@
+/* Copyright (c) 2013-2018, EPFL/Blue Brain Project
+ *                          Jonas Karlsson <jonas.karlsson@epfl.ch>
+ *
+ * This file is part of Brion <https://github.com/BlueBrain/Brion>
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License version 3.0 as published
+ * by the Free Software Foundation.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE.  See the GNU Lesser General Public License for more
+ * details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this library; if not, write to the Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ */
+
+#include "nodes.h"
+
+#include <HighFive/include/highfive/H5DataSet.hpp>
+#include <HighFive/include/highfive/H5File.hpp>
+
+#include <iostream>
+#include <memory>
+
+namespace brion
+{
+static uint32_ts get_id_list_helper(const HighFive::File& file,
+                                    const std::string& population,
+                                    const std::string& dataset)
+{
+    uint32_ts list;
+    const HighFive::Group group = file.getGroup("/nodes/" + population);
+    const HighFive::DataSet ds = group.getDataSet(dataset);
+    ds.read(list);
+    return list;
+}
+//////////////////////////////////////////////////////////////////////////////
+
+struct Nodes::Impl
+{
+    std::unique_ptr<HighFive::File> file;
+};
+
+Nodes::Nodes(const URI& uri)
+    : impl(new Nodes::Impl())
+{
+    const std::string path = uri.getPath();
+
+    try
```

Comment:
Do you think you really need to catch the exception to rethrow?
It's less lines of code if you do something like:

    : impl(new Node::Impl(uri.getPath())

And then you can also save the pointer in Nodes::Impl

It's also not consistent that you are catching the HighFive exception here, but not in any other function.